sources/api/apiserver/src/server/mod.rs

```diff
@@ -379,6 +424,38 @@ async fn reboot() -> Result<HttpResponse> {
 
 // Helpers for handler methods called by the router
 
+fn call_thar_be_updates(args: Vec<String>) -> Result<HttpResponse> {
+    let status = Command::new("/usr/bin/thar-be-updates")
+        .args(args)
+        .status()
+        .context(error::UpdateDispatcher)?;
+    if status.success() {
+        Ok(HttpResponse::NoContent().finish())
+    } else {
+        let exit_status = match status.code() {
+            Some(code) => code,
+            None => status.signal().unwrap_or(1),
+        };
+        Err(match_update_dispatcher_errors(exit_status, None))
+    }
+}
+
+fn match_update_dispatcher_errors(exit_status: i32, stderr: Option<Vec<u8>>) -> Error {
+    let stderr = match stderr {
+        Some(s) => s,
+        None => vec![],
+    };
+    let error = FromPrimitive::from_i32(exit_status);
+    match error {
+        Some(TbuErrorStatus::LockHeld) => error::Error::UpdateLockHeld {},
+        Some(TbuErrorStatus::DisallowedCmd) => error::Error::UpdateActionNotAllowed {},
+        Some(TbuErrorStatus::UpdateDoesNotExist) => error::Error::UpdateMissing {},
+        Some(TbuErrorStatus::NoStagedImage) => error::Error::NoStagedImage {},
+        // other errors
+        _ => error::Error::UpdateError {},
```

Comment:
```suggestion
        _ => error::Error::UpdateError,
```
Don't need the curlies to build structs with no fields.

(Elsewhere too.)